docs/csharp/how-to/compare-strings.md

```diff
@@ -0,0 +1,86 @@
+
+[!code-csharp-interactive[Sorting a list of strings](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#7)]
+
+Once sorted, the list of strings can be searched using a binary search. The following sample shows how to search the sorted listed using the same comparison function. The local function `ShowWhere` shows where the sought text is or would be:
+
+[!code-csharp-interactive[csProgGuideStrings#11](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#8)]
+
+Collection classes such as <xref:System.Collections.Hashtable?displayProperty=nameWithType>, <xref:System.Collections.Generic.Dictionary%602?displayProperty=nameWithType>, and <xref:System.Collections.Generic.List%601?displayProperty=nameWithType> have constructors that take a <xref:System.StringComparer?displayProperty=nameWithType> parameter when the type of the elements or keys is `string`. In general, you should use these constructors whenever possible, and specify either <xref:System.StringComparer.Ordinal?displayProperty=nameWithType> or <xref:System.StringComparer.OrdinalIgnoreCase?displayProperty=nameWithType>.
+
+## Comparing across cultures
+
+The following example shows several aspects of comparing strings across cultures. A string is initialized to the phrase "They dance in the street." in German. Linguistically (in Windows), "ss" is equal to the German essetz: 'ß' character in both "en-US" and "de-DE" cultures. This sample stores <xref:System.Globalization.CultureInfo> for the current culture. The original culture can be set and retrieved on the current thread object. The comparisons are performed using the <xref:System.StringComparison.CurrentCulture> value to ensure a culture-specific comparison.
+
+The following example shows how to compare strings using the <xref:System.String?displayProperty=nameWithType> methods that take a <xref:System.StringComparison> enumeration. The <xref:System.String.CompareTo%2A?displayProperty=nameWithType> instance methods are not used here, because overloads take a <xref:System.StringComparison>. Restrict your use of the <xref:System.String.CompareTo%2A?displayProperty=nameWithType> instance methods to when you want an alphabetic ordering. The host for the interactive REPL uses the **invariant culture**, which does not have a name. The sample uses the <xref:System.Globalization.CultureInfo.EnglishName?displayName=nameWithType> property to display a human readable string for this culture. 
+
+[!code-csharp-interactive[Comparing strings across cultures](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#9)]
```

Comment:
There's a problem with this code in the REPL. Here's the output:
     First sentence is <Sie tanzen auf der Straße.>
     Second sentence is <Sie tanzen auf der Strasse.>
     Comparing in Invariant Language (Invariant Country) returns 1.
     Comparing in de-DE returns 1.
     The two strings are not equal.
The issues:
- For some reason, en-US was not set as the current culture, even though it should have been.
- Even the de-DE comparison fails. The basic problem may be that the REPL is running on a non-Windows machine. Windows supplies the culture-sensitive character tables for .NET Core (I think). I have no idea of where culture-sensitive character comes from on non-Windows systems, though. @tarekgh, might that be the source of the problem?

Reply:
The string comparisons and searching operations depend on the OS, the framework doesn't carry the collation tables and it just call the underlying OS for such operations. so the behavior is fully depended on the OS.